heartbeat/monitors/wrappers/summarizer/summarizer.go

```diff
@@ -0,0 +1,173 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package summarizer
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/elastic/beats/v7/heartbeat/eventext"
+	"github.com/elastic/beats/v7/heartbeat/monitors/jobs"
+	"github.com/elastic/beats/v7/heartbeat/monitors/stdfields"
+	"github.com/elastic/beats/v7/heartbeat/monitors/wrappers/monitorstate"
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+type Summarizer struct {
+	rootJob        jobs.Job
+	contsRemaining uint16
+	mtx            *sync.Mutex
+	jobSummary     *JobSummary
+	checkGroup     string
+	stateTracker   *monitorstate.Tracker
+	sf             stdfields.StdMonitorFields
+	retryDelay     time.Duration
+}
+
+type JobSummary struct {
+	Id           int
+	Attempt      uint16                   `json:"attempt"`
+	MaxAttempts  uint16                   `json:"max_attempts"`
+	FinalAttempt bool                     `json:"final_attempt"`
+	Up           uint16                   `json:"up"`
+	Down         uint16                   `json:"down"`
+	Status       monitorstate.StateStatus `json:"status"`
+	RetryGroup   string                   `json:"retry_group"`
+}
+
+func NewSummarizer(rootJob jobs.Job, sf stdfields.StdMonitorFields, mst *monitorstate.Tracker) *Summarizer {
+	uu, err := uuid.NewV1()
+	if err != nil {
+		logp.L().Errorf("could not create v1 UUID for retry group: %s", err)
+	}
+	return &Summarizer{
+		rootJob:        rootJob,
+		contsRemaining: 1,
+		mtx:            &sync.Mutex{},
+		jobSummary:     NewJobSummary(1, sf.MaxAttempts, uu.String()),
+		checkGroup:     uu.String(),
+		stateTracker:   mst,
+		sf:             sf,
+		// private property, but can be overriden in tests to speed them up
+		retryDelay: time.Second,
+	}
+}
+
+var id int
```

Comment:
Doesn't seem like it's used effectively? If it's needed, it should be atomic uint imo.